mne/decoding/time_gen.py

```diff
@@ -203,16 +224,31 @@ def predict(self, epochs):
         if not hasattr(self, 'estimators_'):
             raise RuntimeError('Please fit models before trying to predict')
 
-        # clean attributes
+        # Check predict mode
+        if self.predict_mode not in ['cross-validation', 'mean-prediction']:
+            raise ValueError('`predict_mode` must be a str, "mean-prediction" '
+                             'or "cross-validation"')
+
+        # Check that training cv and predicting cv match
+        if self.predict_mode == 'cross-validation':
+            n_est_cv = [len(estimator) for estimator in self.estimators_]
```

Comment:
There's one estimator per fold per training time. `estimators_` stores all of them in a list of list.

`n_est_cv` is a list of the number of estimator in each time point. It should be contains identical values == `k_folds`.